test/tests/utilities.js

```diff
@@ -272,5 +272,89 @@ describe("Zotero.Utilities", function() {
 			
 			assert.isUndefined(cslJSON.PMID, 'field labels are case-sensitive');
 		});
+		it("should parse particles in creator names", function() {
+			let creators = [
+				{
+					// No particles
+					firstName: 'John',
+					lastName: 'Smith',
+					creatorType: 'author',
+					expect: {
+						given: 'John',
+						family: 'Smith'
+					}
+				},
+				{
+					// dropping and non-dropping
+					firstName: 'Jean de',
+					lastName: 'la Fontaine',
+					creatorType: 'author',
+					expect: {
+						given: 'Jean',
+						"dropping-particle": 'de',
+						"non-dropping-particle": 'la',
+						family: 'Fontaine'
+					}
+				},
+				{
+					// only non-dropping
+					firstName: 'Vincent',
+					lastName: 'van Gogh',
+					creatorType: 'author',
+					expect: {
+						given: 'Vincent',
+						"non-dropping-particle": 'van',
+						family: 'Gogh'
+					}
+				},
+				{
+					// only dropping
+					firstName: 'Alexander von',
+					lastName: 'Humboldt',
+					creatorType: 'author',
+					expect: {
+						given: 'Alexander',
+						"dropping-particle": 'von',
+						family: 'Humboldt'
+					}
+				},
+				{
+					// institutional author
+					lastName: 'Jean de la Fontaine',
+					creatorType: 'author',
+					fieldMode: 1,
+					expect: {
+						literal: 'Jean de la Fontaine'
+					}
+				},
+				{
+					// protected last name
+					firstName: 'Jean de',
+					lastName: '"la Fontaine"',
+					creatorType: 'author',
+					expect: {
+						given: 'Jean de',
+						family: 'la Fontaine'
+					}
```

Comment:
this mirrors current citeproc behavior (AFAICT), but it's somewhat odd. As can be seen, protecting the last name prevents parsing of the first name as well. To be honest, I'm not sure when parsing the first name is actually important (is dropping particle ever used independently?), but if it is, this isn't right.